src/lib/mbtiles.ts

```diff
@@ -0,0 +1,185 @@
+import { Metadata } from '@mapbox/mbtiles'
+import SphericalMercator from '@mapbox/sphericalmercator'
+import { Database as DatabaseInstance } from 'better-sqlite3'
+
+import { TileJSON } from './tilejson'
+import { generateId } from './utils'
+
+type ValidMBTilesFormat = TileJSON['format']
+
+export const VALID_MBTILES_FORMATS: ValidMBTilesFormat[] = [
+  'pbf',
+  'png',
+  'jpg',
+  'webp',
+]
+
+export function isValidMBTilesFormat(
+  format: string
+): format is ValidMBTilesFormat {
+  return VALID_MBTILES_FORMATS.includes(format as ValidMBTilesFormat)
+}
+
+export function mbTilesToTileJSON(mbTilesDb: DatabaseInstance): TileJSON {
+  const metadata = extractMBTilesMetadata(mbTilesDb)
+
+  return {
+    ...metadata,
+    id: generateId(),
+    // TODO: are we strictly supporting this version of the tilejson spec?
+    tilejson: '2.2.0',
+    // Technically not compliant with spec
```

Comment:
Can you add a note about why it is not compliant and why we are including it, for future readers :)

Reply:
Addressed via https://github.com/digidem/mapeo-mapserver/pull/19/commits/59f53b7cb6c2003415f1cc4607a4f661a6d8245d